### PR TITLE
add the intel-ipsec-mb package

### PR DESCRIPTION
**Pull request by snambakam**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Adding the intel-ipsec-mb package.

###### Change Log  

Adding the intel-ipsec-mb package.

###### Does this affect the toolchain?  

NO

### CODE REVIEW

File: SPECS/intel-ipsec-mb/intel-ipsec-mb.spec [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+%files
+
+%license LICENSE
+%doc README ReleaseNotes.txt
+
+%{_libdir}/libIPSec_MB.so.%{fullversion}
+%{_libdir}/libIPSec_MB.so.%{major}
+
+%{_mandir}/man7/libipsec-mb.7.gz
```
Comment (oliviacrain) [inline SPECS/intel-ipsec-mb/intel-ipsec-mb.spec:101]: Nit: This is more easily parsed by the eye without the newlines.

```suggestion
%files
%license LICENSE
%doc README ReleaseNotes.txt
%{_libdir}/libIPSec_MB.so.%{fullversion}
%{_libdir}/libIPSec_MB.so.%{major}
%{_mandir}/man7/libipsec-mb.7.gz
```

File: SPECS/intel-ipsec-mb/intel-ipsec-mb.spec [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+https://github.com/intel/%{githubname}
+
+%prep
+%autosetup -n %{githubfull}
+
+%if 0%{?rhel} && 0%{?rhel} < 8
+%ldconfig_post
+
+%ldconfig_postun
+%endif
```
Comment (oliviacrain) [inline SPECS/intel-ipsec-mb/intel-ipsec-mb.spec:73]: In general, we should try to remove these distro checks- these specs will probably never be used outside of a Mariner context. Usually, for Fedora specs, this means taking the Fedora-specific behavior and removing RHEL-specific behaviors.

In this case though, we desire the RHEL behavior of calling the ldconfig macros. [The short explanation](https://fedoraproject.org/wiki/Changes/Removing_ldconfig_scriptlets) is that newer versions of Fedora use file triggers in the `glibc` package to handle calling `ldconfig` pre/post install. We don't have that, so we need to specify this behavior manually.

In summary- this `%if` block can be removed and we can add the following equivalent line below the `%install` section:

`%ldconfig_scriptlets`

File: SPECS/intel-ipsec-mb/intel-ipsec-mb.spec [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+BuildRequires:  nasm >= 2.14
+ExclusiveArch:  x86_64
+
+%description
+IPSEC cryptography library optimized for Intel Architecture
+
+%package -n intel-ipsec-mb-devel
+Summary:        IPSEC cryptography library optimized for Intel Architecture
+Group:          Development/Tools
+Requires:       %{name}%{?_isa} = %{version}-%{release}
```
Comment (oliviacrain) [inline SPECS/intel-ipsec-mb/intel-ipsec-mb.spec:57]: Nit: If I'm remembering correctly, Fedora uses the `%{?_isa}` macro so that 64-bit machines don't pull the 32-bit version of the packages in [multilib environments](https://fedoraproject.org/wiki/PackagingDrafts/MultilibTricks). This isn't a concern for Mariner. It's mostly harmless for us to keep in, but I personally prefer to remove these macros.

File: SPECS/intel-ipsec-mb/intel-ipsec-mb.spec [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+# Versions numbers
+BuildRequires:  gcc >= 4.8.3
+BuildRequires:  make
+BuildRequires:  nasm >= 2.14
+ExclusiveArch:  x86_64
+
+%description
+IPSEC cryptography library optimized for Intel Architecture
+
+%package -n intel-ipsec-mb-devel
```
Comment (oliviacrain) [inline SPECS/intel-ipsec-mb/intel-ipsec-mb.spec:54]: Nit: Could macro-ize the base package name here.

```suggestion
%package -n %{name}-devel
```

File: SPECS/intel-ipsec-mb/intel-ipsec-mb.spec [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (oliviacrain) [inline SPECS/intel-ipsec-mb/intel-ipsec-mb.spec:44]: Unless this is the license of the actual spec file, this license can be removed. Packaging the license in the `%files` section is sufficient.

File: SPECS/intel-ipsec-mb/intel-ipsec-mb.spec [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+%if 0%{?rhel} && 0%{?rhel} < 8
+%ldconfig_post
+
+%ldconfig_postun
+%endif
+
+%build
+cd lib
+make EXTRA_CFLAGS='%{optflags}' %{?_smp_mflags}
```
Comment (oliviacrain) [inline SPECS/intel-ipsec-mb/intel-ipsec-mb.spec:77]: Nit: The `%make_build` macro inserts the `%{?_smp_mflags}` flags and enables more verbose logging, which can be useful if the package build breaks for some reason.

```suggestion
%make_build EXTRA_CFLAGS='%{optflags}'
```

**Review by oliviacrain — COMMENTED** [review]
Looks good overall! Just needs a bit of tweaking around a couple of areas of concern, and I have a few nits.

**Review by oliviacrain — APPROVED** [review]

**Review by oliviacrain — CHANGES_REQUESTED** [review]
Revoking the approval- just realized I forgot to check one thing. The `%ldconfig_scriptlets` line needs to be added between `%install` and `%files` sections, so that `ldconfig` is called on pre/post install.

**Review by oliviacrain — APPROVED** [review]
